src/pages/index.tsx

```diff
@@ -49,11 +115,27 @@ const PortalHomePage = ({ pastEvents, upcomingEvents, liveEvents, attendances }:
 
 export default PortalHomePage;
 
-const getServerSidePropsFunc: GetServerSideProps = async ({ req, res }) => {
+const getServerSidePropsFunc: GetServerSideProps = async ({ req, res, query }) => {
   const authToken = CookieService.getServerCookie(CookieType.ACCESS_TOKEN, { req, res });
-  const events = await EventAPI.getAllEvents();
-  const attendances = await EventAPI.getAttendancesForUser(authToken);
 
+  let checkInResponse = null;
+  if (typeof query.code === 'string') {
+    // If a check-in code is specified, first check in to that event.
+    checkInResponse = await attendEvent({ token: authToken, attendanceCode: query.code });
```

Comment:
I can do that, but the intention is to run the check-in API code first since the other API calls (user + attendance) may be different depending on the outcome of the check-in API call